tsdb/engine/tsm1/data_file.go

```diff
@@ -631,10 +630,11 @@ func (d *indirectIndex) Delete(key string) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
 
+	kb := []byte(key)
 	var offsets []int32
 	for _, offset := range d.offsets {
 		_, indexKey, _ := readKey(d.b[offset:])
-		if key == indexKey {
+		if bytes.Compare(kb, indexKey) == 0 {
```

Comment:
Nitpicky, but I think `bytes.Equal` is more idiomatic that `bytes.Compare(...) == 0`.